ouroboros-consensus-cardano/tools/db-analyser/Main.hs

```diff
@@ -1,9 +1,14 @@
-{-# LANGUAGE FlexibleContexts #-}
-{-# LANGUAGE NamedFieldPuns   #-}
-{-# LANGUAGE RecordWildCards  #-}
+{-# LANGUAGE FlexibleContexts    #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE RankNTypes          #-}
+{-# LANGUAGE RecordWildCards     #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 -- | Database analyse tool.
 module Main (main) where
 
+import           Codec.CBOR.Decoding (Decoder)
+import           Codec.Serialise (Serialise (decode))
```

Comment:
Is it easy enough to use `fromCBOR` instead of `decode`? That's generally preferable and would avoid having to add the `serialise` package to `build-depends`.

Reply:
Not sure, we don't have `FromCBOR` defined for `ExtLedgerState` 

```
    • Could not deduce (cardano-binary-1.5.0:Cardano.Binary.FromCBOR.FromCBOR
                          (ExtLedgerState blk))
        arising from a use of ‘fromCBOR’
```

and this is now `LederDB` reads those snapshots

```
initFromDisk LgrDbArgs { lgrHasFS = hasFS, .. }
             replayTracer
             immutableDB = wrapFailure (Proxy @blk) $ do
    (_initLog, db, replayed) <-
      LedgerDB.initLedgerDB
        replayTracer
        lgrTracer
        hasFS
        decodeExtLedgerState'
        decode
        (configLedgerDb lgrTopLevelConfig)
        lgrGenesis
        (streamAPI immutableDB)
    return (db, replayed)
  where
    ccfg = configCodec lgrTopLevelConfig

    decodeExtLedgerState' :: forall s. Decoder s (ExtLedgerState blk)
    decodeExtLedgerState' = decodeExtLedgerState
                              (decodeDisk ccfg)
                              (decodeDisk ccfg)
                              (decodeDisk ccfg)
```

We could potentially provide `FromCBOR` instance. But that would be a bit out of scope for this ticket (we can create new ticket though).